smdebug/tensorflow/base_hook.py

```diff
@@ -449,6 +454,11 @@ def _make_numpy_array(tensor_value):
         Convert the tensor value into a numpy array.
         Here it's already numpy array
         """
+        if (
+            isinstance(tensor_value, tf.Variable) or isinstance(tensor_value, tf.Tensor)
+        ) and hasattr(tensor_value, "numpy"):
```

Comment:
good catch!
in eager mode, it does: https://github.com/tensorflow/tensorflow/blob/5c7cf8b83bff16d1ff276914a79b81c907bc7c8d/tensorflow/python/framework/ops.py#L937.
in non-eager mode, we have to use tf.keras.backend.eval(tensor_value) [will have to check if there's a performance impact due to this]

Reply:
AFAIK that method makes another session run call. Yeah there'll be perf impact if we use the eval method

Reply:
we dont have to go this path for non-eager. in TF 2.x, default is keras eager. keras non-eager should be the same as TF 1.15 default keras. i've changed the code to fall back to TF 1.15 keras in case of non-eager mode. all tests pass.
will quickly check the perf numbers